shared/src/pool_fetching.rs

```diff
@@ -24,13 +24,27 @@ const POOL_SWAP_GAS_COST: usize = 60_000;
 /// `reserve_a` refers to the reserve of the excluded token.
 type RelativeReserves = (U256, U256, H160);
 
+/// The state of the chain at which information should be retrieved.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub enum Block {
+    /// The most recent state. This is on a best effort basis so that for example a cache can still
+    /// return results that are slightly out of date.
+    Latest,
```

Comment:
Let's not call it the same as BlockNumber::Latest means something quite different (the comment clarifies it, but I'm afraid some future maintainer might shortcut their understanding). Maybe Recent?

Reply:
Renamed to Recent.